File: 0228-summary-ranges/0228-summary-ranges.cpp
class Solution {
public:
    vector<string> summaryRanges(vector<int>& nums) {
        int size = nums.size();
        if(size == 0) return {};
        int f = nums[0], l = nums[size-1];
        vector<string> result;
        vector<int> temp;
        for(int i=0;i<size-1;i++){
            if(nums[i] + 1 == nums[i+1]){
                temp.push_back(nums[i]);
            }
            else {
                if(temp.size() == 0){
                    result.push_back(to_string(nums[i]));
                }
                else {
                    result.push_back(to_string(temp[0]) + "->" + to_string(nums[i]));
                }
                temp.clear();
            }
        }
        if(temp.size() == 0){
            result.push_back(to_string(nums[size-1]));
        }
        else {
            result.push_back(to_string(temp[0]) + "->" + to_string(nums[size-1]));
        }
        return result;
    }
};

File: 0228-summary-ranges/0228-summary-ranges.java
class Solution {
    public List<String> summaryRanges(int[] nums) {
        if(nums.length == 0) return new ArrayList<>();
        List<String> res = new ArrayList<>();
        int start = 0;
        for(int i=0;i<nums.length-1;i++){
            if(nums[i] + 1 != nums[i+1]){
                if(start == i) {
                    res.add(String.format("%d", nums[start]));
                }else {
                    res.add(String.format("%d->%d", nums[start],nums[i]));
                }
                start = i+1;
            }
        }
        if(start != nums.length-1){
            res.add(String.format("%d->%d", nums[start],nums[nums.length-1]));
        }else {
            res.add(String.format("%d", nums[start]));
        }
        return res;
    }
}